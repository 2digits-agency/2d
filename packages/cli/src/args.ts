import yargs from 'yargs';

export const argsParser = yargs
  .command('init', 'Initialize a new project.', (args) => {
    return args.positional('dir', { type: 'string' });
  })
  .help()
  .strict()
  .parse();
