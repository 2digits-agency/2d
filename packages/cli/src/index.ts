import cli from './cli';

export default function main(...args: string[]) {
  cli.parse(args, { from: 'user' });
}
