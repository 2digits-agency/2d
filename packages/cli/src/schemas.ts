import fs from 'fs-extra';
import pathe from 'pathe';
import { z } from 'zod';

export const appName = z
  .string()
  .nonempty()
  .regex(/^[\da-z~-][\d._a-z~-]*$/, {
    message: 'Please check https://docs.npmjs.com/cli/v9/configuring-npm/package-json#name',
  });

export const moduleEnum = z.enum([
  'web',
  'storybook',
  'trpc',
  'tailwind',
  'swagger-sdk',
  'gql-codegen',
]);

const VALID_PROJECT_DIRECTORY_SAFE_LIST = new Set([
  '.DS_Store',
  '.gitkeep',
  '.gitattributes',
  '.gitignore',
  '.npmignore',
  'LICENSE',
  'Thumbs.db',
]);

function isEmpty(dirPath: string) {
  try {
    // Throws if the directory does not exist.
    const files = fs.readdirSync(dirPath);

    // If the directory is not empty, check if it contains only safe files.
    return files.every((content) => VALID_PROJECT_DIRECTORY_SAFE_LIST.has(content));
  } catch {
    // If the directory does not exist, it is considered empty.
    return true;
  }
}

export const appPath = z.string().nonempty().refine(isEmpty, 'Directory has to be empty');

export const moduleName = (root: string) =>
  appName.refine(
    (name) => isEmpty(pathe.resolve(root, 'packages', name)),
    'Directory has to be empty',
  );
