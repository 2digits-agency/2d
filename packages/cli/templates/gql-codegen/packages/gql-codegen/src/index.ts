import { GraphQLClient } from 'graphql-request';

export * from './gql';

export const client = new GraphQLClient(
  'https://swapi-graphql.netlify.app/.netlify/functions/index',
);
