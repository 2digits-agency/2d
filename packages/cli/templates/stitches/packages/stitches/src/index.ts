export * from './stitches';
export * from './types';
export * from './helpers';
