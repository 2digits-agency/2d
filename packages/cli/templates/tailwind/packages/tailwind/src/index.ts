import { findUpSync } from 'find-up';
import pathe from 'pathe';
import type { Config } from 'tailwindcss';

const workspaceMeta = findUpSync('pnpm-workspace.yaml');
const workspaceRoot = workspaceMeta ? pathe.dirname(workspaceMeta) : '../../';
const packagesDir = pathe.resolve(workspaceRoot, 'packages');

export const tailwindConfig = {
  content: ['./src/**/*.{ts,tsx}', `${packagesDir}/*/src/**/*.{ts,tsx}`],
} satisfies Config;
