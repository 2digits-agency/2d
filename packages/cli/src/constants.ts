import { fileURLToPath } from 'node:url';

import pathe from 'pathe';

import base from '../templates/base/package.json';
import gqlCodegen from '../templates/gql-codegen/packages/gql-codegen/package.json';
import module from '../templates/module/package.json';
import storybook from '../templates/storybook/apps/storybook/package.json';
import swaggerSdk from '../templates/swagger-sdk/packages/swagger-sdk/package.json';
import tailwind from '../templates/tailwind/packages/tailwind/package.json';
import trpc from '../templates/trpc/packages/trpc/package.json';
import web from '../templates/web/apps/web/package.json';

// With the move to TSUP as a build tool, this keeps path routes in other files (installers, loaders, etc) in check more easily.
// Path is in relation to a single index.js file inside ./dist
const __filename = fileURLToPath(import.meta.url);

export const PKG_DIST = pathe.dirname(__filename);

export const PKG_ROOT = pathe.join(PKG_DIST, '..');

export const TEMPLATE_DIR = pathe.join(PKG_ROOT, 'templates');

export const templates = {
  base,
  web,
  tailwind,
  trpc,
  ['swagger-sdk']: swaggerSdk,
  storybook,
  ['gql-codegen']: gqlCodegen,
  module,
} as const;

export type Template = keyof typeof templates;
