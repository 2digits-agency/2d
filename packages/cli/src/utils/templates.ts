import chalk from 'chalk';
import { createConsola } from 'consola';
import fs from 'fs-extra';
import ignore from 'ignore';
import pathe from 'pathe';

import { Spinner } from '@2digits/log';

import base from '../../public/templates/base/package.json';
import web from '../../public/templates/web/apps/web/package.json';
import { PKG_DIST } from '../constants';
import { onCancel } from '../helpers';

const consola = createConsola({ defaults: { tag: 'utils/templates' } });

export const templates = {
  base,
  web,
} as const;

export type Template = keyof typeof templates;

export async function copyTemplate(template: Template, path: string) {
  const sourceDir = pathe.join(PKG_DIST, 'templates', template);

  consola.debug('sourceDir', sourceDir);

  const ignoreFile = pathe.join(sourceDir, '.gitignore');

  consola.debug('ignoreFile', ignoreFile);

  const ig = ignore();

  if (await fs.exists(ignoreFile)) {
    const ignoreContents = await fs.readFile(ignoreFile, { encoding: 'utf8' });

    const ignoreLines = ignoreContents.split('\n').filter((line) => ignore.isPathValid(line));

    consola.debug('ignoreLines', ignoreLines);

    ig.add(ignoreLines);
  }

  const spinner = new Spinner();

  const destPath = pathe.relative(process.cwd(), path);

  consola.debug('destPath', destPath);

  spinner.start(`Copying ${chalk.bold(`${template}`)} template to ${chalk.bold(destPath)}...`);

  try {
    await fs.copy(sourceDir, path, {
      errorOnExist: true,
      filter(src, dest) {
        const relative = pathe.relative(sourceDir, src);

        consola.debug('filtering: ' + relative);
        consola.trace('src: ' + src);
        consola.trace('dest: ' + dest);

        spinner.suffixText = relative;

        return !(relative && ig.ignores(relative));
      },
    });

    spinner.success(`Copied ${chalk.bold(`${template} template`)} to ${chalk.bold(destPath)}`);
  } catch (error) {
    consola.fatal(error);

    spinner.fail(
      chalk.red(`Failed to copy ${chalk.bold(`${template} template`)} to ${chalk.bold(destPath)}`),
    );

    onCancel();
  }
}
