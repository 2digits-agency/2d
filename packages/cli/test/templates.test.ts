import consola from 'consola';
import mock from 'mock-fs';
import { afterEach, beforeEach, describe, expect, expectTypeOf, it, test } from 'vitest';

import type { Template } from '../src/constants';
import { templates } from '../src/constants';
import { copyTemplate } from '../src/utils/templates';
import { getCombinations, getMockFsFiles, getTemplateFiles } from './testUtils';
import { moduleEnum } from '../src/schemas';

const templateNames = Object.keys(templates) as Template[];

const combinations = getCombinations(templateNames);

describe('copyTemplate', async () => {
  const mockFileSystem = await getTemplateFiles();

  beforeEach(() => {
    mock(mockFileSystem);

    consola.wrapStd();
    consola.pauseLogs();
  });

  afterEach(() => {
    mock.restore();

    consola.resumeLogs();
    consola.restoreAll();
  });

  for (const combination of combinations) {
    const name = combination.join(' + ');

    test(name, async () => {
      for (const template of combination) {
        await copyTemplate(template, './test/path');
      }

      const files = await getMockFsFiles();

      mock.restore();

      consola.debug(files);

      expect(files).toMatchSnapshot();
    });
  }
});

describe('types', () => {
  it('should export a correct Templates type', () => {
    expectTypeOf(moduleEnum.options).items.toMatchTypeOf<Template>();
  });
});
