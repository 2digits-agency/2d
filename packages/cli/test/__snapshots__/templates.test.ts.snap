// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`copyTemplate > base + web > ./test/path/.editorconfig 1`] = `
"# editorconfig.org
root = true

[*]
indent_style = space
indent_size = 2
"
`;

exports[`copyTemplate > base + web > ./test/path/.prettierignore 1`] = `
"pnpm-lock.yaml
public
"
`;

exports[`copyTemplate > base + web > ./test/path/_2d_.gitignore 1`] = `
"# dependencies
node_modules
.pnp
.pnp.js
package-lock.json
yarn.lock

# tsup
dist/

# misc
.DS_Store
*.pem

# debug
npm-debug.log*
yarn-debug.log*
yarn-error.log*
.pnpm-debug.log*

# local env files
.env
.env*.local

# turbo
.turbo

# vercel cli
.vercel

# next.js
.next/
out/
build
next-env.d.ts
"
`;

exports[`copyTemplate > base + web > ./test/path/_2d_pnpm-workspace.yaml 1`] = `
"packages:
  - 'apps/*'
  - 'packages/*'
"
`;

exports[`copyTemplate > base + web > ./test/path/_2d_turbo.json 1`] = `
"{
  \\"globalDependencies\\": [\\"**/.env\\", \\"tsconfig.json\\"],
  \\"pipeline\\": {
    \\"build\\": { \\"dependsOn\\": [\\"^build\\"], \\"outputs\\": [\\"dist/**\\", \\"build/**\\"] },
    \\"dev\\": { \\"cache\\": false, \\"persistent\\": true },
    \\"lint\\": { \\"cache\\": false },
    \\"lint-fix\\": { \\"cache\\": false },
    \\"types\\": { \\"dependsOn\\": [\\"^types\\"], \\"cache\\": false }
  }
}
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/next.config.mjs 1`] = `
"import bundleAnalyzer from '@next/bundle-analyzer';
import { withSentryConfig } from '@sentry/nextjs';

const shouldAnalyze = process.env.ANALYZE === 'true';

/** @type {(config?: import('next').NextConfig) => import('next').NextConfig} */
const withBundleAnalyzer = bundleAnalyzer({
  enabled: shouldAnalyze,
});

/** @type {import('next').NextConfig} */
const config = {
  reactStrictMode: true,

  /** We run eslint as a separate task in CI */
  eslint: { ignoreDuringBuilds: true },
};

export default withSentryConfig(
  shouldAnalyze ? withBundleAnalyzer(config) : config,
  {},
  { hideSourceMaps: true },
);
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/package.json 1`] = `
"{
  \\"name\\": \\"@mod/web\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"types\\": \\"tsc --noEmit\\"
  },
  \\"dependencies\\": {
    \\"@sentry/cli\\": \\"^2.17.5\\",
    \\"@sentry/nextjs\\": \\"^7.50.0\\",
    \\"next\\": \\"^13.3.4\\",
    \\"react\\": \\"^18.2.0\\",
    \\"react-dom\\": \\"^18.2.0\\"
  },
  \\"devDependencies\\": {
    \\"@2digits/tsconfig\\": \\"^0.1.1\\",
    \\"@babel/core\\": \\"^7.21.8\\",
    \\"@next/bundle-analyzer\\": \\"^13.3.4\\",
    \\"@types/node\\": \\"^18.16.3\\",
    \\"@types/react\\": \\"^18.2.0\\",
    \\"@types/react-dom\\": \\"^18.2.1\\",
    \\"eslint\\": \\"^8.39.0\\",
    \\"typescript\\": \\"^5.0.4\\"
  }
}
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/sentry.client.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/sentry.config.ts 1`] = `
"import * as Sentry from '@sentry/nextjs';

function initSentry() {
  const SENTRY_DSN = process.env.SENTRY_DSN ?? process.env.NEXT_PUBLIC_SENTRY_DSN;

  if (!SENTRY_DSN) {
    console.warn('No Sentry DSN found, check your environment variables');

    return;
  }

  Sentry.init({
    enabled: process.env.NODE_ENV !== 'development',
    dsn: SENTRY_DSN,
  });
}

initSentry();
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/sentry.edge.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/sentry.server.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/src/pages/_app.tsx 1`] = `
"import type { AppProps } from 'next/app';

function App({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />;
}

export default App;
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/src/pages/index.tsx 1`] = `
"export default function Home() {
  return (
    <div>
      <h1>Hello, world!</h1>
    </div>
  );
}
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/tsconfig.json 1`] = `
"{
  \\"extends\\": \\"@2digits/tsconfig/tsconfig.json\\",
  \\"include\\": [\\"next-env.d.ts\\", \\"**/*.ts\\", \\"**/*.tsx\\", \\"next.config.mjs\\", \\".next/types/**/*.ts\\"],
  \\"compilerOptions\\": {
    \\"incremental\\": true,
    \\"baseUrl\\": \\"./src\\",
    \\"module\\": \\"esnext\\",
    \\"jsx\\": \\"preserve\\",
    \\"plugins\\": [
      {
        \\"name\\": \\"next\\"
      }
    ],
    \\"moduleResolution\\": \\"node\\"
  },
  \\"exclude\\": [\\"node_modules\\"]
}
"
`;

exports[`copyTemplate > base + web > ./test/path/apps/web/turbo.json 1`] = `
"{
  \\"extends\\": [\\"//\\"],
  \\"pipeline\\": {
    \\"build\\": {
      \\"env\\": [\\"SENTRY_DSN\\", \\"NEXT_PUBLIC_SENTRY_DSN\\", \\"ANALYZE\\", \\"NODE_ENV\\", \\"VERCEL_URL\\"],
      \\"outputs\\": [\\".next/**\\", \\"!.next/cache/**\\"]
    }
  }
}
"
`;

exports[`copyTemplate > base + web > ./test/path/package.json 1`] = `
"{
  \\"name\\": \\"@mod/base\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"scripts\\": {},
  \\"dependencies\\": {
    \\"@2digits/eslint-config\\": \\"^0.3.0\\",
    \\"@2digits/prettier-config\\": \\"^0.3.0\\",
    \\"@2digits/tsconfig\\": \\"^0.1.1\\",
    \\"@manypkg/cli\\": \\"^0.20.0\\",
    \\"@types/node\\": \\"^18.16.3\\",
    \\"eslint\\": \\"^8.39.0\\",
    \\"plop\\": \\"^3.1.2\\",
    \\"prettier\\": \\"^2.8.8\\",
    \\"turbo\\": \\"^1.9.3\\",
    \\"typescript\\": \\"^5.0.4\\"
  },
  \\"packageManager\\": \\"pnpm@8.4.0\\",
  \\"engines\\": {
    \\"node\\": \\">=18.0.0\\"
  },
  \\"manypkg\\": {
    \\"ignoredRules\\": [
      \\"INTERNAL_MISMATCH\\"
    ]
  },
  \\"pnpm\\": {
    \\"neverBuiltDependencies\\": [
      \\"core-js\\",
      \\"core-js-compat\\",
      \\"core-js-pure\\",
      \\"cypress\\"
    ]
  },
  \\"prettier\\": \\"@2digits/prettier-config\\",
  \\"eslintConfig\\": {
    \\"extends\\": \\"@2digits\\"
  }
}
"
`;

exports[`copyTemplate > base + web > ./test/path/tsconfig.json 1`] = `
"{
  \\"extends\\": \\"@2digits/tsconfig/tsconfig.json\\"
}
"
`;

exports[`copyTemplate > base > ./test/path/.editorconfig 1`] = `
"# editorconfig.org
root = true

[*]
indent_style = space
indent_size = 2
"
`;

exports[`copyTemplate > base > ./test/path/.prettierignore 1`] = `
"pnpm-lock.yaml
public
"
`;

exports[`copyTemplate > base > ./test/path/_2d_.gitignore 1`] = `
"# dependencies
node_modules
.pnp
.pnp.js
package-lock.json
yarn.lock

# tsup
dist/

# misc
.DS_Store
*.pem

# debug
npm-debug.log*
yarn-debug.log*
yarn-error.log*
.pnpm-debug.log*

# local env files
.env
.env*.local

# turbo
.turbo

# vercel cli
.vercel

# next.js
.next/
out/
build
next-env.d.ts
"
`;

exports[`copyTemplate > base > ./test/path/_2d_pnpm-workspace.yaml 1`] = `
"packages:
  - 'apps/*'
  - 'packages/*'
"
`;

exports[`copyTemplate > base > ./test/path/_2d_turbo.json 1`] = `
"{
  \\"globalDependencies\\": [\\"**/.env\\", \\"tsconfig.json\\"],
  \\"pipeline\\": {
    \\"build\\": { \\"dependsOn\\": [\\"^build\\"], \\"outputs\\": [\\"dist/**\\", \\"build/**\\"] },
    \\"dev\\": { \\"cache\\": false, \\"persistent\\": true },
    \\"lint\\": { \\"cache\\": false },
    \\"lint-fix\\": { \\"cache\\": false },
    \\"types\\": { \\"dependsOn\\": [\\"^types\\"], \\"cache\\": false }
  }
}
"
`;

exports[`copyTemplate > base > ./test/path/package.json 1`] = `
"{
  \\"name\\": \\"@mod/base\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"scripts\\": {},
  \\"dependencies\\": {
    \\"@2digits/eslint-config\\": \\"^0.3.0\\",
    \\"@2digits/prettier-config\\": \\"^0.3.0\\",
    \\"@2digits/tsconfig\\": \\"^0.1.1\\",
    \\"@manypkg/cli\\": \\"^0.20.0\\",
    \\"@types/node\\": \\"^18.16.3\\",
    \\"eslint\\": \\"^8.39.0\\",
    \\"plop\\": \\"^3.1.2\\",
    \\"prettier\\": \\"^2.8.8\\",
    \\"turbo\\": \\"^1.9.3\\",
    \\"typescript\\": \\"^5.0.4\\"
  },
  \\"packageManager\\": \\"pnpm@8.4.0\\",
  \\"engines\\": {
    \\"node\\": \\">=18.0.0\\"
  },
  \\"manypkg\\": {
    \\"ignoredRules\\": [
      \\"INTERNAL_MISMATCH\\"
    ]
  },
  \\"pnpm\\": {
    \\"neverBuiltDependencies\\": [
      \\"core-js\\",
      \\"core-js-compat\\",
      \\"core-js-pure\\",
      \\"cypress\\"
    ]
  },
  \\"prettier\\": \\"@2digits/prettier-config\\",
  \\"eslintConfig\\": {
    \\"extends\\": \\"@2digits\\"
  }
}
"
`;

exports[`copyTemplate > base > ./test/path/tsconfig.json 1`] = `
"{
  \\"extends\\": \\"@2digits/tsconfig/tsconfig.json\\"
}
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/next.config.mjs 1`] = `
"import bundleAnalyzer from '@next/bundle-analyzer';
import { withSentryConfig } from '@sentry/nextjs';

const shouldAnalyze = process.env.ANALYZE === 'true';

/** @type {(config?: import('next').NextConfig) => import('next').NextConfig} */
const withBundleAnalyzer = bundleAnalyzer({
  enabled: shouldAnalyze,
});

/** @type {import('next').NextConfig} */
const config = {
  reactStrictMode: true,

  /** We run eslint as a separate task in CI */
  eslint: { ignoreDuringBuilds: true },
};

export default withSentryConfig(
  shouldAnalyze ? withBundleAnalyzer(config) : config,
  {},
  { hideSourceMaps: true },
);
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/package.json 1`] = `
"{
  \\"name\\": \\"@mod/web\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"types\\": \\"tsc --noEmit\\"
  },
  \\"dependencies\\": {
    \\"@sentry/cli\\": \\"^2.17.5\\",
    \\"@sentry/nextjs\\": \\"^7.50.0\\",
    \\"next\\": \\"^13.3.4\\",
    \\"react\\": \\"^18.2.0\\",
    \\"react-dom\\": \\"^18.2.0\\"
  },
  \\"devDependencies\\": {
    \\"@2digits/tsconfig\\": \\"^0.1.1\\",
    \\"@babel/core\\": \\"^7.21.8\\",
    \\"@next/bundle-analyzer\\": \\"^13.3.4\\",
    \\"@types/node\\": \\"^18.16.3\\",
    \\"@types/react\\": \\"^18.2.0\\",
    \\"@types/react-dom\\": \\"^18.2.1\\",
    \\"eslint\\": \\"^8.39.0\\",
    \\"typescript\\": \\"^5.0.4\\"
  }
}
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/sentry.client.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/sentry.config.ts 1`] = `
"import * as Sentry from '@sentry/nextjs';

function initSentry() {
  const SENTRY_DSN = process.env.SENTRY_DSN ?? process.env.NEXT_PUBLIC_SENTRY_DSN;

  if (!SENTRY_DSN) {
    console.warn('No Sentry DSN found, check your environment variables');

    return;
  }

  Sentry.init({
    enabled: process.env.NODE_ENV !== 'development',
    dsn: SENTRY_DSN,
  });
}

initSentry();
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/sentry.edge.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/sentry.server.config.ts 1`] = `
"import './sentry.config';
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/src/pages/_app.tsx 1`] = `
"import type { AppProps } from 'next/app';

function App({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />;
}

export default App;
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/src/pages/index.tsx 1`] = `
"export default function Home() {
  return (
    <div>
      <h1>Hello, world!</h1>
    </div>
  );
}
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/tsconfig.json 1`] = `
"{
  \\"extends\\": \\"@2digits/tsconfig/tsconfig.json\\",
  \\"include\\": [\\"next-env.d.ts\\", \\"**/*.ts\\", \\"**/*.tsx\\", \\"next.config.mjs\\", \\".next/types/**/*.ts\\"],
  \\"compilerOptions\\": {
    \\"incremental\\": true,
    \\"baseUrl\\": \\"./src\\",
    \\"module\\": \\"esnext\\",
    \\"jsx\\": \\"preserve\\",
    \\"plugins\\": [
      {
        \\"name\\": \\"next\\"
      }
    ],
    \\"moduleResolution\\": \\"node\\"
  },
  \\"exclude\\": [\\"node_modules\\"]
}
"
`;

exports[`copyTemplate > web > ./test/path/apps/web/turbo.json 1`] = `
"{
  \\"extends\\": [\\"//\\"],
  \\"pipeline\\": {
    \\"build\\": {
      \\"env\\": [\\"SENTRY_DSN\\", \\"NEXT_PUBLIC_SENTRY_DSN\\", \\"ANALYZE\\", \\"NODE_ENV\\", \\"VERCEL_URL\\"],
      \\"outputs\\": [\\".next/**\\", \\"!.next/cache/**\\"]
    }
  }
}
"
`;
